### PR TITLE
Shooter Trigger Subsystem

### PR DESCRIPTION
**Pull request by riversry**
## Justification

This subsystem is intended to be a single motor that controls the holding and feeding of notes in the turret shooter.
It should be able to load the note from the intake subsystem, run at a very slow speed to prevent the note from slipping out, and feed the shooter when firing.
There may also be a photogate for detecting held notes.

## Implementaion

The majority of the code is a copy of the intake subsystem because their functionalities are very similar.

## Testing Done

No testing has been done so far.

### CODE REVIEW

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -41,6 +42,7 @@ public class RobotContainer {
 
   // The robot's subsystems and commands are defined here...
   private final IntakeSubsystem intakeSubsystem = new IntakeSubsystem();
+  private final ShooterTriggerSubsystem shooterTriggerSubsystem = new ShooterTriggerSubsystem();
```
Comment (lohengrin332) [inline src/main/java/frc/robot/RobotContainer.java:45]: Honest question: why instantiate this here, as opposed to in the constructor?

**Comment by lohengrin332** [comment]
I'll approve this PR once conflicts are resolved. Please let me know when you are ready.

File: src/main/java/frc/robot/RobotContainer.java (hunk 1 above)
Comment (zbrotherton) [reply src/main/java/frc/robot/RobotContainer.java]: Mistype copied from intake

**Comment by azvanderpas** [comment]
Will approve once conflicts are resolved

File: src/main/java/frc/robot/subsystems/TriggerSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    return buildCommand(TriggerConstants.LOAD_PERCENTAGE);
+  }
+
+  public Command getHoldCommand() {
+    return buildCommand(TriggerConstants.HOLD_PERCENTAGE);
+  }
+
+  public Command getShootCommand() {
+    return buildCommand(TriggerConstants.SHOOT_PERCENTAGE);
+  }
```
Comment (azvanderpas) [inline src/main/java/frc/robot/subsystems/TriggerSubsystem.java:33]: Add .withName(...) to the end of the returned Commands so we can track them in logs

**Review by caseyjbrotherton — APPROVED** [review]

**Pull request merged**